### PR TITLE
RTA-372 Call bakers when they are present. Use ajax and /j endpoint

### PR DESCRIPTION
**Pull request by melgenek**
This pr still uses the embedded ajax and pixel implementations. I'll create a separate pr to make them pluggable.

Related prs:
- https://github.com/LiveIntent/Prebid.js/pull/3
- https://github.com/LiveIntent/LiveConnectBuilder/pull/25

### CODE REVIEW

**Comment by jankoulaga** [comment]
@melgenek this looks cool! Are we extracting the ajax to have the same signature as prebid in a separate PR?

**Comment by melgenek** [comment]
@jankoulaga yes. I'll create a separate pr for the extraction. I want to release a dedicated version of lc2 builder with the changes from this pr to keep the changes in each version granular.

File: src/utils/json.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Parses the string as json. If the string is not a valid json, then an empty json is returned.
+ * @param json a json string
+ * @returns {{}}
+ */
+export function safeParseJson (json) {
+  try {
+    return JSON.parse(json)
+  } catch (e) {
+    return {}
```
Comment (jankoulaga) [inline src/utils/json.js:10]: I guess we'd want to emit this error

File: src/pixel/sender.js [hunk 2]
```diff
@@ -13,25 +16,50 @@ export function PixelSender (liveConnectConfig, onload, presend) {
    * @param {StateWrapper} state
    * @private
    */
-  function _send (state) {
+  function _sendAjax (state) {
+    _sendState(state, 'j', uri => {
```
Comment (jankoulaga) [inline src/pixel/sender.js:20]: DOn't we need the onload here as well?

File: src/pixel/sender.js [hunk 3]
```diff
@@ -13,25 +16,50 @@ export function PixelSender (liveConnectConfig, onload, presend) {
...
+   * @private
+   */
+  function _sendPixel (state) {
+    _sendState(state, 'p', uri => {
+      sendPixel(uri, onload)
+    })
+  }
+
+  function _sendState (state, endpoint, makeCall) {
     if (state.sendsPixel()) {
```
Comment (gipeshka) [inline src/pixel/sender.js:42]: does this one mean that it sends only pixel? Or is it now `send in general`?

**Review by gipeshka — COMMENTED** [review]
Does this PR contain parts of referer fix?

File: src/pixel/sender.js (hunk 2 above)
Comment (melgenek) [reply src/pixel/sender.js:20]: it was one line below. I moved the bakers' calls into a separate function to make the code a bit more clear. And now the ajax callback calls the "onload" and "callBakers" https://github.com/LiveIntent/live-connect/pull/25/files#diff-51925b852a9fa34aa4d658fa7cd30c4dR22-R23

File: src/utils/json.js (hunk 1 above)
Comment (melgenek) [reply src/utils/json.js]: I removed the json parsing utility and catch all the errors that happen in the /baker calls https://github.com/LiveIntent/live-connect/pull/25/files#diff-51925b852a9fa34aa4d658fa7cd30c4dR35. The error is emitted now.

File: src/pixel/sender.js (hunk 3 above)
Comment (melgenek) [reply src/pixel/sender.js:48]: "sendsPixel" now means "send via any medium".But I don't know how to rename this and I am fine keeping it as "send pixel".

**Comment by melgenek** [comment]
@gipeshka there was a pr into a branch that is not a master when I merged it the changes were shown for the referrer as well.
I pushed the merge with master, this pr is only about the ajax+/bakers calls now.

**Review by jankoulaga — APPROVED** [review]

**Comment by jankoulaga** [comment]
@gipeshka can you check this please?

**Review by gipeshka — APPROVED** [review]

**Pull request merged**